src/sidebar/services/frame-sync.js

```diff
@@ -235,17 +236,30 @@ export class FrameSyncService {
       });
     };
 
-    const discovery = new Discovery(window, { server: true });
-    discovery.startDiscovery(
-      (source, origin, token) =>
-        this._bridge.createChannel({ source, origin, token }),
-      [
-        // Ping the host frame which is in most cases also the only guest.
-        window.parent,
-      ]
-    );
     this._bridge.onConnect(addFrame);
 
+    // Listen for messages from new guest frames that want to connect.
+    //
+    // The message will include a `MessagePort` to use for communication with
+    // the guest. Communication with the host currently relies on the host
+    // frame also always being a guest frame.
+    this._window.addEventListener('message', e => {
```

Comment:
This listener is not removed. It may not be necessary, but it is always a good idea. Is there a way to cleanly destroy services, if so `ListenerCollection` utility could be handy.

Reply:
If the listener is not destroyed, this class will never be able to be garbage collected, except if the window itself is destroyed.

Reply:
`FrameSyncService` currently provide any way to destroy/dispose it. This is true of all services in the Hypothesis client as they are instantiated only once when the application starts. In unit tests we pass in a fake window which will be destroyed at the end of each test. The same applies to other services passed into the constructor.